### PR TITLE
Administrator ACLs

### PR DESCRIPTION
**Pull request by mmadariaga**
Added PublicEntity and AdministratorRelPublicEntity into the ORM

#### Type Of Change 
- [ ] Small bug fix
- [X] New feature or enhancement
- [ ] Breaking change

#### Checklist:
- [X] Commits are named and have tag following [commit rules](https://github.com/irontec/ivozprovider/blob/bleeding/doc/dev/en/commits.md)
- [X] Commits are split per component (schema, web/admin, kamusers, agis, ..)
- [X] Changes have been tested locally
- [ ] Fixes an existing issue (Fixes #XXXX)

### CODE REVIEW

File: schema/app/DoctrineMigrations/Version20200210145817.php [hunk 1]
```diff
@@ -0,0 +1,408 @@
...
+    /**
+     * @param Schema $schema
+     */
+    public function up(Schema $schema)
+    {
+        // this up() migration is auto-generated, please modify it to your needs
+        $this->abortIf($this->connection->getDatabasePlatform()->getName() !== 'mysql',
+            'Migration can only be executed safely on \'mysql\'.');
+
+        $clientEntities = $this->getBrandPublicEntities();
```
Comment (mmadariaga) [inline schema/app/DoctrineMigrations/Version20200210145817.php]: $brandEntities

File: schema/app/DoctrineMigrations/Version20200210145817.php [hunk 2]
```diff
@@ -0,0 +1,408 @@
...
+        $insertQuery = 'INSERT INTO `PublicEntities` (`iden`, `fqdn`, `brand`, `name_en`, `name_es`) VALUES (%s, %s, 1, %s, %s) ON DUPLICATE KEY UPDATE `brand` = 1';
```
Comment (mmadariaga) [inline schema/app/DoctrineMigrations/Version20200210145817.php]: make this multi line

File: schema/app/DoctrineMigrations/Version20200210145818.php [hunk 3]
```diff
@@ -0,0 +1,228 @@
...
+    /**
+     * @param Schema $schema
+     */
+    public function up(Schema $schema)
+    {
+        // this up() migration is auto-generated, please modify it to your needs
+        $this->abortIf($this->connection->getDatabasePlatform()->getName() !== 'mysql',
+            'Migration can only be executed safely on \'mysql\'.');
+
+        $clientEntities = $this->getPlatformPublicEntities();
```
Comment (mmadariaga) [inline schema/app/DoctrineMigrations/Version20200210145818.php]: $platformEntities

File: schema/app/DoctrineMigrations/Version20200210145818.php [hunk 4]
```diff
@@ -0,0 +1,228 @@
...
+        $insertQuery = 'INSERT INTO `PublicEntities` (`iden`, `fqdn`, `platform`, `name_en`, `name_es`) VALUES (%s, %s, 1, %s, %s) ON DUPLICATE KEY UPDATE `platform` = 1';
```
Comment (mmadariaga) [inline schema/app/DoctrineMigrations/Version20200210145818.php]: multiline

File: library/Ivoz/Provider/Domain/Service/AdministratorRelPublicEntity/UpdateAcls.php [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+     * @return void
+     */
+    public function execute(AdministratorInterface $admin)
+    {
+        $isNew = $admin->isNew();
+        if ($isNew) {
+            return;
+        }
+
+        $restrictedhasNotChanged = !$admin->hasChanged(
```
Comment (mmadariaga) [inline library/Ivoz/Provider/Domain/Service/AdministratorRelPublicEntity/UpdateAcls.php]: $restricted**H**as

File: schema/tests/Provider/AdministratorRelPublicEntity/AdministratorRelPublicEntityRepositoryTest.php [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+            ->getRepository(AdministratorRelPublicEntity::class);
+
+        $admin = $this->getAdmin();
+
+        $administratorRelPublicEntityRepository
+            ->setWritePermissions($admin);
+        $this
+            ->resetChangelog();
+
+        $administratorRelPublicEntityRepository
```
Comment (mmadariaga) [inline schema/tests/Provider/AdministratorRelPublicEntity/AdministratorRelPublicEntityRepositoryTest.php]: assert affectedRows

File: schema/tests/Provider/AdministratorRelPublicEntity/AdministratorRelPublicEntityRepositoryTest.php [hunk 7]
```diff
@@ -0,0 +1,111 @@
...
+            ->getRepository(AdministratorRelPublicEntity::class);
+
+        $admin = $this->getAdmin();
+
+        $administratorRelPublicEntityRepository
+            ->setReadOnlyPermissions($admin);
+        $this
+            ->resetChangelog();
+
+        $administratorRelPublicEntityRepository
```
Comment (mmadariaga) [inline schema/tests/Provider/AdministratorRelPublicEntity/AdministratorRelPublicEntityRepositoryTest.php]: assert affectedRows

**Comment by ironArt3mis** [comment]
Rebase completed.

**Pull request merged**